File: python-algorithm/leetcode/problem_258.py
"""258. Add Digits
https://leetcode.com/problems/add-digits/
"""


def add_digits(num: int) -> int:
    while num > 9:
        cur = 0
        while num > 0:
            num, a = divmod(num, 10)
            cur += a
        num = cur
    return num

File: go-algo/leetcode/problem258.go
package main

// 258.
//

func addDigits(num int) int {
    next_num := 0
    for num > 9 {
        a, b := num / 10, num % 10
        next_num += b
        num = a
        if num < 10 {
            num += next_num
            next_num = 0
        }
    }
    return num
}

